src/V3SenExprBuilder.h

```diff
@@ -29,7 +29,7 @@
 
 class SenExprBuilder final {
     // STATE
-    AstScope* const m_scopep;  // The scope
+    AstScope* m_scopep;  // The scope
```

Comment:
I don't think this change is needed, you never write it outside the constructor.

Reply:
Because C++ is the way it is, keeping `const` there deletes `SenExprBuilder`'s copy constructor (and move constructor). Other solutions include:
- Manually writing a custom copy constructor/move constructor/reinit function that copies/clears all `SenExprBuilder` fields (of which there's quite a few),
- Keeping the `SenExprBuilder` on the heap and reallocating it for each scope,
- Calling the destructor and placement `new` on it for each scope, e.g.
  ```cpp
  senExprBuilder.~SenExprBuilder();
  new (&senExprBuilder) SenExprBuilder{m_scopep}; 
  ```
  That may not be a terrible solution if we wrap it in some sort of `reconstruct(...)` utility function, I guess. Though that doesn't work with your suggestion that we need a restorer here.